caddy_chatbot/src/caddy_core/components.py

```diff
@@ -415,21 +417,42 @@ def reword_advisor_message(message: str) -> str:
         5. Extract any mentioned dates, locations, or monetary amounts.
         6. Identify any legal terms or concepts mentioned. 
     Provide your response in a structured format with clear headings for each category of extracted information. 
-    If any category is not applicable or no relevant information is found skip it. 
+    If any category is not applicable or no relevant information is found, skip it. 
     Remember to focus only on extracting factual information without adding any interpretation or advice. 
     
     Input:
     {message}
     
     Extracted Information:
     """
-    response = llm.invoke(prompt)
-    return response.content    
+    
+    reworded_response = llm.invoke(reword_prompt)
+
+    classification_prompt = ChatPromptTemplate.from_template(
+        """There are 3 types of questions asked:
+        Research: Seeking information or how-to guides.
+        Client: Specific scenarios often involving detailed client situations.
+        General: Broad or non-specific requests for information or help.
+        Based on the following extracted information, 
+        determine the question type and return the question type only.
+        
+        Extracted Information:
+        {reworded_response}
+        
+        Type of Question:
+        """
+    )
```

Comment:
I think these two prompts could be combined. you can include the classification logic into the rewording prompt. It would save us having to call the llm twice. I checked and annoyingly I don't think chatbedrock has the ability to output results in json. So you may want to instruct the model to do the classification in a fairly verbose way so it gets picked up by a semantic router